File: code/player.py
# player.py
import pygame
from maze import N, S, E, W, DX, DY

class Player:
    def __init__(self, x, y, color=(0, 0, 255), speed=5): 
        # ... (init, start_move, stop_move, update 함수는 이전과 동일) ...
        self.grid_x = x 
        self.grid_y = y
        self.pixel_x = x * 60 # (임시값, main에서 덮어씀)
        self.pixel_y = y * 60 
        self.color = color
        self.speed = speed 

        self.moving_direction = None 
        self.target_pixel_x = self.pixel_x
        self.target_pixel_y = self.pixel_y
    
    # ... (start_move, stop_move, update 함수는 변경 없음) ...
    def start_move(self, grid, direction, cell_size):
        if self.pixel_x != self.target_pixel_x or self.pixel_y != self.target_pixel_y:
            return
        cell = grid[self.grid_y][self.grid_x]
        if cell & direction: 
            self.moving_direction = direction
            self.grid_x += DX[direction]
            self.grid_y += DY[direction]
            self.target_pixel_x = self.grid_x * cell_size
            self.target_pixel_y = self.grid_y * cell_size
        else:
            self.moving_direction = None 
    def stop_move(self):
        self.moving_direction = None
    def update(self):
        if self.moving_direction is not None:
            if self.pixel_x < self.target_pixel_x:
                self.pixel_x = min(self.pixel_x + self.speed, self.target_pixel_x)
            elif self.pixel_x > self.target_pixel_x:
                self.pixel_x = max(self.pixel_x - self.speed, self.target_pixel_x)
            if self.pixel_y < self.target_pixel_y:
                self.pixel_y = min(self.pixel_y + self.speed, self.target_pixel_y)
            elif self.pixel_y > self.target_pixel_y:
                self.pixel_y = max(self.pixel_y - self.speed, self.target_pixel_y)
            if self.pixel_x == self.target_pixel_x and self.pixel_y == self.target_pixel_y:
                self.moving_direction = None 

    def draw(self, surface, cell_size):
        draw_x = self.pixel_x + cell_size // 2
        draw_y = self.pixel_y + cell_size // 2
        
        pygame.draw.circle(surface, self.color, (draw_x, draw_y), 12)

File: design/assets/player.js
// assets/player.js - 최종 수정본 (순수한 원형 캐릭터 + 랜덤 색상)

import { HARD, selectDifficultyFromKey } from "./difficulty.js";

// 🆕 랜덤 16진수 색상 코드를 생성하는 함수
const generateRandomColor = () => {
    // # 다음에 6자리의 랜덤 16진수를 생성합니다.
    return '#' + Math.floor(Math.random() * 16777215).toString(16).padStart(6, '0');
};

export class Player {
    constructor(
        gridX,
        gridY,
        cellSize,
        // 🚨 color 매개변수 제거
        baseSpeed = 2
    ) {
        this.grid_x = gridX;
        this.grid_y = gridY;
        this.cellSize = cellSize;
        
        // 🚨 🆕 랜덤 색상 설정
        this.color = generateRandomColor(); 
        
        this.speed = baseSpeed;
        this.baseSpeed = baseSpeed;

        // 1. 📏 난이도에 따른 플레이어 크기 조정 (난이도 로직 유지)
        const dKey = new URLSearchParams(window.location.search).get("d");
        const difficulty = selectDifficultyFromKey(dKey); 

        let margin;
        if (difficulty === HARD) {
            margin = 2; // HARD 모드: 최소 마진 (2px)
        } else {
            margin = Math.max(2, Math.round(cellSize * 0.15)); // EASY 모드: 마진 적용
        }

        this.size = this.cellSize - margin;

        // 현재 픽셀 위치 (셀 중앙)
        this.pixel_x = gridX * cellSize + cellSize / 2;
        this.pixel_y = gridY * cellSize + cellSize / 2;
        this.target_x = this.pixel_x;
        this.target_y = this.pixel_y;

        this.moving_direction = null;

        // DOM element 생성 (순수한 몸체만)
        this.el = document.createElement("div");
        this.el.className = "player";
        
        // 🚨 🆕 랜덤 색상 적용
        this.el.style.background = this.color;
        
        this.el.style.width = `${this.size}px`;
        this.el.style.height = `${this.size}px`;

        this.updateDom();
    }

    // DOM 위치 업데이트 (셀 중앙 기준)
    updateDom() {
        this.el.style.left = `${this.pixel_x}px`;
        this.el.style.top = `${this.pixel_y}px`;
    }

    // --- 이동 로직 유지 ---
    
    // 이동 시작 시도
    start_move(grid, dir, cellSize) {
        if (this.moving_direction !== null) return;

        const nx = this.grid_x + dir.dx;
        const ny = this.grid_y + dir.dy;

        // 범위 체크 및 벽 체크 로직은 동일
        if (ny < 0 || ny >= grid.length) return;
        if (nx < 0 || nx >= grid[0].length) return;

        const cell = grid[this.grid_y][this.grid_x];
        if ((cell & dir.bit) === 0) {
            return;
        }

        this.grid_x = nx;
        this.grid_y = ny;

        this.target_x = this.grid_x * cellSize + cellSize / 2;
        this.target_y = this.grid_y * cellSize + cellSize / 2;

        this.moving_direction = dir;
    }

    // 매 프레임 호출
    update() {
        if (this.moving_direction === null) {
            this.updateDom();
            return;
        }

        const dx = this.target_x - this.pixel_x;
        const dy = this.target_y - this.pixel_y;
        const dist = Math.hypot(dx, dy);

        if (dist <= this.speed) {
            this.pixel_x = this.target_x;
            this.pixel_y = this.target_y;
            this.moving_direction = null;
        } else {
            this.pixel_x += (dx / dist) * this.speed;
            this.pixel_y += (dy / dist) * this.speed;
        }

        this.updateDom();
    }

    draw() {}
}